bool initialized_
uint32 data